openfe/utils/logging_control.py

```diff
@@ -0,0 +1,175 @@
+import logging
+from abc import ABC, abstractmethod
+
+
+class BaseLogFilter(ABC):
+    """Base class for log filters that handle string or list of strings.
+
+    Parameters
+    ----------
+    strings : str or list of str
+        String(s) to use in the filter logic
+    """
+
+    def __init__(self, strings: str | list[str]) -> None:
+        if isinstance(strings, str):
+            strings = [strings]
+        self.strings: list[str] = strings
+
+    @abstractmethod
+    def filter(self, record: logging.LogRecord) -> bool:
+        """Filter method to be implemented by subclasses.
+
+        Parameters
+        ----------
+        record : logging.LogRecord
+            Log record to filter/modify
+
+        Returns
+        -------
+        bool
+            True to allow the record, False to block it
+        """
+        ...
+
+
+class MsgIncludesStringFilter(BaseLogFilter):
+    """Logging filter to silence specific log messages.
+
+    See https://docs.python.org/3/library/logging.html#filter-objects
+
+    Parameters
+    ----------
+    strings : str or list of str
+        If string(s) match in log messages (substring match) then the log record
+        is suppressed
+    """
+
+    def filter(self, record: logging.LogRecord) -> bool:
+        """Filter log records that contain any of the specified strings.
+
+        Parameters
+        ----------
+        record : logging.LogRecord
+            Log record to filter
+
+        Returns
+        -------
+        bool
+            False if the record should be blocked, True if it should be logged
+        """
+        for string in self.strings:
+            if string in record.msg:
+                return False
+        return True
+
+
+class AppendMsgFilter(BaseLogFilter):
+    """Logging filter to append a message to a specific log message.
+
+    See https://docs.python.org/3/library/logging.html#filter-objects
+
+    Parameters
+    ----------
+    strings : str or list of str
+        Suffix text(s) to append to log messages
+    """
+
+    def __init__(self, strings: str | list[str]) -> None:
+        super().__init__(strings)
+        # Rename for clarity in this context
+        self.suffixes = self.strings
+
+    def filter(self, record: logging.LogRecord) -> bool:
+        """Append suffix to log record message.
+
+        Parameters
+        ----------
+        record : logging.LogRecord
+            Log record to modify
+
+        Returns
+        -------
+        bool
+            Always True to allow the record to be logged
+        """
+        for suffix in self.suffixes:
+            # Only modify if not already appended (idempotent)
+            if not record.msg.endswith(suffix):
+                record.msg = f"{record.msg}{suffix}"
+        return True
+
+
+class LogControl:
```

Comment:
I'm not super fond of doing a class w/ staticmethods instead of just methods in the module (i.e. this class could have just been 3 methods), but it's really not something I feel that strongly about - will default to @atravitz's opinion.

Reply:
I agree with Irfan - a class can be helpful for organization, but I think the usage:

```python
from openfe.utils import logging_control

logging_control.append_logger(...

```
is just as clear, so in this case there's no need for a class. @mikemhenry is there a benefit to this that we're missing?

Reply:
I remember now, at one point I had things working so that we could delete filters if needed so we needed a `LogControl` state but after I got rid of that I never thought about how it is stateless.